util/argo/resource_tracking.go

```diff
@@ -151,18 +156,26 @@ func (rt *resourceTracking) SetAppInstance(un *unstructured.Unstructured, key, v
 		if len(val) > LabelMaxLength {
 			val = val[:LabelMaxLength]
 		}
-		return argokube.SetAppInstanceLabel(un, key, val)
+		err = argokube.SetAppInstanceLabel(un, key, val)
+		if err != nil {
+			return fmt.Errorf("failed to set app instance label: %w", err)
```

Comment:
Actually, do you know the name of the app?

Reply:
I'll add that to some more places where this is logged...